### DISCUSSION TITLE
constructor anonymous json variable

### DISCUSSION THREAD

**Discussion by howudodat** [nlohmann/json#2434] [Q&A] [answered]
The following code wont even generate any assembly, it completes skips over the constructor
```
class base {
	void init (nlohmann::json j) { }
	string toString() { return "YAY"; }
};

class child : public base {
	child(nlohmann::json& j) {
		init(j);
	}
};
	string s1 = "some json";
	child b3(nlohmann::json(s1));   <<<<<< This line produces no assembly code
```
adding
`	cout << b3.toString() << endl;
`
Produces a compiler error:
request for member ‘toString’ in ‘b3’, which is of non-class type ‘child(nlohmann::json)’ {aka ‘child(nlohmann::basic_json<>)’}

Any idea what I might be doing wrong?

**Comment by howudodat** [comment, marked as answer]
I cant explain why it didn't produce any assembly, however here is the correct way to pas this around:

```
class base {
public:
	void init (nlohmann::json j) { }
	string toString() { return "YAY"; }
};

class child : public base {
public:
	child(const nlohmann::json& j) {
		init(j);
	}
};

	child c(nlohmann::json::parse(s1));   <<<< use parse() and dont create a json string object
```

see also https://github.com/nlohmann/json/discussions/2435

**Comment by gregmarr** [comment]
This is why it didn't produce any assembly: https://github.com/nlohmann/json/issues/2427#issuecomment-707158212
